### PR TITLE
CMP API fidesClearCookie to load CMP without preferences on refresh 

### PR DESCRIPTION
**Pull request by Kelsey-Ethyca**
Closes https://ethyca.atlassian.net/browse/PROD-1976 

### Description Of Changes

On page reload the fides_consent cookie is deleted and the expected experience is shown as if the user has never interacted with the CMP before 

### Code Changes

* [ ] add fidesCleanCookie as a Fides.option, env setting, and query param 
* [ ] update getOrMakeFidesCookie to check if fidesCleanCookie is true, if so delete the current fides_consent cookie and create a new default cookie 

### Steps to Confirm

Setup
* [x] run fides `turbo run dev`
* [x] run fidesplus `nox -s "build(slim)" "dev(slim)" -- dev` 
* [x] setup privacy notices and experiences by adding GVL systems to the admin-ui and enabling all notices and experiences 
* [x] nav to fides.js demo page http://localhost:3000/fides-js-demo.html

Test Non-TCF
* [x] verify fidesClearCookie deletes the fides_consent cookie on refresh as if it is the first time navigating to the page even after saving preferences http://localhost:3000/fides-js-demo.html?geolocation=ca-qc&fides_clear_cookie=true
* [x] verify the fides_cookie is still present on refresh when fidesClearCookie is not enabled http://localhost:3000/fides-js-demo.html?geolocation=ca-qc
* [x] verify the cookie is written to the browser is written to the browser after save no matter fidesClearCookie setting 
* [x] verify the cookie is deleted after setting fidesClearCookie is true even if preferences have been save previously with fidesClearCookie false 

Test TCF 
* [x] verify fidesClearCookie deletes the fides_consent cookie on refresh as if it is the first time navigating to the page even after saving preferences http://localhost:3000/fides-js-demo.html?geolocation=eea&fides_clear_cookie=true
* [x] verify the fides_cookie is still present on refresh when fidesClearCookie is not enabled http://localhost:3000/fides-js-demo.html?geolocation=eea
* [x] verify the cookie is written to the browser is written to the browser after save no matter fidesClearCookie setting 
* [x] verify the cookie is deleted after setting fidesClearCookie is true even if preferences have been save previously with fidesClearCookie false 

Test Embed Mode 
* [x] verify fidesClearCookie deletes the fides_consent cookie on refresh as if it is the first time navigating to the page even after saving preferences http://localhost:3000/fides-js-demo.html?geolocation=ca-qc&fides_clear_cookie=true&fides_embed=true
* [x] verify the fides_cookie is still present on refresh when fidesClearCookie is not enabled http://localhost:3000/fides-js-demo.html?geolocation=ca-qc&fides_embed=true
* [x] verify the cookie is written to the browser is written to the browser after save no matter fidesClearCookie setting 
* [x] verify the cookie is deleted after setting fidesClearCookie is true even if preferences have been save previously with fidesClearCookie false 

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Comment by Kelsey-Ethyca** [comment]
UAT https://www.loom.com/share/6d8850e55da242ac958fc26a3ff0910d?sid=f2e37ecd-b757-454b-a98d-988ef192a20e

File: clients/privacy-center/app/server-environment.ts [hunk 1]
```diff
@@ -369,6 +371,9 @@ export const loadPrivacyCenterEnvironment =
       FIDES_PRIMARY_COLOR: process.env.FIDES_PRIVACY_CENTER__FIDES_PRIMARY_COLOR
         ? process.env.FIDES_PRIVACY_CENTER__FIDES_PRIMARY_COLOR
         : null,
+      FIDES_CLEAR_COOKIE: process.env.FIDES_CLEAR_COOKIE
+        ? process.env.FIDES_PRIVACY_CENTER__FIDES_PRIMARY_COLOR === "true"
```
Comment (galvana) [inline clients/privacy-center/app/server-environment.ts:375]:
```suggestion
        ? process.env.FIDES_CLEAR_COOKIE === "true"
```

File: clients/privacy-center/app/server-environment.ts [hunk 2]
```diff
@@ -369,6 +371,9 @@ export const loadPrivacyCenterEnvironment =
...
+        : false,
```
Comment (galvana) [inline clients/privacy-center/app/server-environment.ts:376]: The environment variable name should be `FIDES_PRIVACY_CENTER__FIDES_CLEAR_COOKIE` to match the current convention
```suggestion
      FIDES_CLEAR_COOKIE: process.env.FIDES_PRIVACY_CENTER__FIDES_CLEAR_COOKIE
        ? process.env.FIDES_PRIVACY_CENTER__FIDES_CLEAR_COOKIE === "true"
        : false,
```

**Review by galvana — CHANGES_REQUESTED** [review]
All the tests passed for the query param use cases but there's an issue loading the env variable, I left an inline comment.

File: CHANGELOG.md [hunk 3]
```diff
@@ -21,6 +21,8 @@ The types of changes are:
 - Fixed bug prevented adding new privacy center translations [#4786](https://github.com/ethyca/fides/pull/4786)
 - Fixed bug where Privacy Policy links would be shown without a configured URL [#4801](https://github.com/ethyca/fides/pull/4801)
 
+### Added 
+- Added CMP API fidesCleanCookie to load CMP without preferences on refresh [#4810](https://github.com/ethyca/fides/pull/4810)
```
Comment (galvana) [inline CHANGELOG.md:25]:
```suggestion
- Added CMP API fidesClearCookie to load CMP without preferences on refresh [#4810](https://github.com/ethyca/fides/pull/4810)
```

**Review by galvana — APPROVED** [review]
Thanks for making the changes, approved! ✅

File: clients/fides-js/src/docs/fides-options.ts [hunk 4]
```diff
@@ -127,4 +127,12 @@ export interface FidesOptions {
    * Defaults to `true`.
    */
   fides_tcf_gdpr_applies: boolean;
+
+  /**
+   * When `true`, shows fides.js overlay UI on load. This deletes the fides_consent cookie as if no preferences have been saved on reload.
```
Comment (NevilleS) [inline clients/fides-js/src/docs/fides-options.ts:132]: nit: This isn't actually accurate though, right? This won't force anything to be shown - it only clears the cookie. There will still be no UI shown if the location doesn't match, `fides_disable_banner` is true, etc.

Anyways, I updated the language here: https://github.com/ethyca/fides/pull/4816/files#diff-f3ea8ea74e0a542fd60a2896722c8e25e63153ebccafacbdc6ae0484d3d95b8aR57

**Pull request merged**